libft/tests/strings/ft_split_skip_delim_test.c

```diff
@@ -0,0 +1,134 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split_skip_delim_test.c                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/23 16:24:51 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/11/02 19:10:44 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../sources/strings/strings.h"
+#include "tests.h"
+
+#define NUM_TEST_CASES 12
+#define MAX_ARGS 16
+
+typedef struct s_string
+{
+	char	*str;
+}	t_string;
+
+typedef struct s_string_array
+{
+	t_string	*strs;
+	int			error;
+	size_t		count;
+}	t_strings;
+
+static t_strings	create_string_array()
+{
+	t_strings	str_array;
+	str_array.count = 12;
+	str_array.error = 0;
+	str_array.strs = ft_calloc_if(sizeof(t_string) * str_array.count, 1);
+	if (str_array.strs == NULL)
+	{
+		str_array.error = 1;
+		return (str_array);
+	}
+	for (int i = 0; i < (int)str_array.count; i++)
+	{
+		str_array.strs[i].str = NULL;
+	}
+	str_array.strs[0].str = strdup("cat ( file1.txt file2.txt ) | grep \"keyword\"");
+	str_array.strs[1].str = strdup("cat < input.txt && echo | grep \"Done\"");
+	str_array.strs[2].str = strdup("grep \"error\" log.txt || echo \"No errors    found\"");
+	str_array.strs[3].str = strdup("gcc main.c -o program >> build.log");
+	str_array.strs[4].str = strdup("find . -name \"*.txt\" | xargs rm -rf || fer && def & echo \"Hello\"");
+	str_array.strs[5].str = strdup("echo \"Compiling...\" && gcc main.c -o program && ./program");
+	str_array.strs[6].str = strdup("mkdir new_dir && cd new_dir && touch file.txt");
+	str_array.strs[7].str = strdup("mv new_dir old_dir");
+	str_array.strs[8].str = strdup("&&");
+	str_array.strs[9].str = strdup("echo \"ls -l && cd ..\"");
+	str_array.strs[10].str = strdup(")(())(");
+	str_array.strs[11].str = strdup("");
+	return (str_array);
+}
+
+static void	free_tokens(char **tokens)
+{
+	if (!tokens)
+		return;
+	for (int i = 0; tokens[i] != NULL; i++)
+	{
+		free(tokens[i]);
+		tokens[i] = NULL;
+	}
+	free(tokens);
+}
+
+static void	free_string_array(t_string *strs, size_t count) 
+{
+	if (!strs)
+		return;
+
+	for (size_t i = 0; i < count; i++) 
+	{
+		free(strs[i].str);
+	}
+	free(strs);
+}
+
+void ft_split_skip_delim_test()
+{
+    t_strings	strings_arr = create_string_array();
+    char	**tokens;
+    int     sz = 0;
+    
+	if (strings_arr.error)
```

Comment:
please keep indents proper :)

Reply:
i deleted this flag because of assert and because it's not a main point of this test. it will be easier to focus only on splitting